python/vowpalwabbit/pyvw.py

```diff
@@ -406,7 +424,13 @@ def predict(self, ec, prediction_type=None):
             new_example = True
 
         # If a string was given, parse it before passing to learner.
-        if isinstance(ec, str):
+        elif isinstance(ec, str):
```

Comment:
Shouldn't this just be if not elif?
```suggestion
        if isinstance(ec, str):
```

Reply:
IMO, it is good to use elif as we are sequentially checking whether it is dict or str or  list